pkgs/development/python-modules/pyobjc-core/default.nix

```
@@ -0,0 +1,111 @@
+# inpired by https://github.com/input-output-hk/lace/blob/0fd166666cc454171811b307050d0815452bea82/nix/lace-blockchain-services/internal/any-darwin.nix#L339
+{
+  lib,
+  pkgs,
+  stdenv,
+  buildPythonPackage,
+  fetchPypi,
+  xcodebuild,
+  cctools,
+  darwin,
+}:
+let
+  appleSDK = darwin.apple_sdk_11_0;
+  inherit (appleSDK) MacOSX-SDK objc4;
+  darwinFrameworks = appleSDK.frameworks;
+
+  apple_libffi = stdenv.mkDerivation {
+    pname = "apple-libffi";
+    inherit (MacOSX-SDK) version;
+    dontUnpack = true;
+    installPhase = ''
+      mkdir -p $out/include $out/lib
+      cp -r ${MacOSX-SDK}/usr/include/ffi $out/include/
+      cp -r ${MacOSX-SDK}/usr/lib/libffi.* $out/lib/
+    '';
+  };
+in
+
+buildPythonPackage rec {
+  pname = "pyobjc_core";
+  version = "10.3.1";
```

```suggestion
  version = "10.3.1";
  pyproject = true;

```
Please add `pyproject = true` and `build-system` to avoid the legacy build.

> pyproject: Whether the pyproject format should be used. As all other formats are deprecated, you are recommended to set this to true. When you do so, pypaBuildHook will be used, and you can add the required build dependencies from build-system.requires to build-system. Note that the pyproject format falls back to using setuptools, so you can use pyproject = true even if the package only has a setup.py. When set to false, you can use the existing [hooks](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#setup-hooks) or provide your own logic to build the package. This can be useful for packages that don't support the pyproject format. When unset, the legacy setuptools hooks are used for backwards compatibility.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters